src/archive.cpp

```diff
@@ -84,18 +85,36 @@ namespace zim
 
   Entry Archive::getEntryByPath(const std::string& path) const
   {
-    log_trace("File::getArticle('" << path << ')');
-    auto r = m_impl->findx(path);
-    if (r.first) {
-      return Entry(m_impl, entry_index_type(r.second));
-    }
-    for (auto ns:{'A', 'I', 'J', '-'}) {
-      auto fallback_path = std::string(1, ns) + "/" + path;
-      r = m_impl->findx(fallback_path);
+    if (m_impl->isNewNamespaceScheme()) {
+      // Get path in user content.
+      auto r = m_impl->findx('C', path);
+      if (r.first) {
+        return Entry(m_impl, entry_index_type(r.second));
+      }
+      try {
+        // Path may come from a already stored from a old zim archive (bookmark),
+        // and so contains a namespace.
+        // We have to adapt the path to use the C namespace.
+        r = m_impl->findx('C', std::get<1>(parseLongPath(path)));
```

Comment:
We have no real choice here.
A old zim file may indeed contains two entries `A/foo` and `I/foo`. The user will bookmark `A/foo`.
The new zim file will have a conflict **at creation** because we want to add twice `foo`in `C` namespace. There are two solutions for this conflict :
- Promote one entry as the "good" entry (`A/foo` -> `C/foo`) and rename the other (either to `C/other_foo_name` or `C/I/foo`.
- Keep both entries in a subdir (`C/A/foo` and `C/I/foo`).

As `A` namespace is the article (main) namespace and `I` is for image, we will probably use the first solution.

At reading of `A/foo` (old path) on the new zim file:
- First solution : We will not find `C/A/foo` and so we will search for `C/foo`.
- Second solution : We will find `C/A/foo`.

In either case, the decision will be taken at zim file creation. At reading, this fallback algorithm seems good to me.
Do you have other user cases where it may fail ? (@kelson42 ?)